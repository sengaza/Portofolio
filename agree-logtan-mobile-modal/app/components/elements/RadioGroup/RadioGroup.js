import React, { Component } from 'react';
import { View } from 'react-native';
import RadioButton from '../RadioButton';
import styles from './RadioGroup.styles';
import { noop } from '../../../utils';

class RadioGroup extends Component {
  render() {
    const { options = [], selectedIndex = 0, spaceBetween = 0, onSelect = noop, disabled } = this.props;
    return (
      <View style={styles.container}>
        {options.map((option, index) => {
          const selected = selectedIndex === index;
          return (
            <RadioButton
              containerStyle={index !== 0 && { marginLeft: spaceBetween }}
              key={option.id}
              index={index}
              name={option.name}
              value={option.id}
              selected={selected}
              onSelect={onSelect}
              labelStyle={disabled && styles.labelStyle}
              circleStyle={disabled && styles.circleStyle}
              selectedStyle={disabled && styles.selectedStyle}
              disabled={disabled}
            />
          );
        })}
      </View>
    );
  }
}

export default RadioGroup;
