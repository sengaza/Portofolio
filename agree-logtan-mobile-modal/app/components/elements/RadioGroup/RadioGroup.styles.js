import { StyleSheet } from 'react-native';
import { COLOR_GREY, COLOR_GREY_60 } from '../../../styles';

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row'
  },
  labelStyle: {
    color: COLOR_GREY
  },
  circleStyle: {
    borderColor: COLOR_GREY
  },
  selectedStyle: {
    backgroundColor: COLOR_GREY_60
  }
});
