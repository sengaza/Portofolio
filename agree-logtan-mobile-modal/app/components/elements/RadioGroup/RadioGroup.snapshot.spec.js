import { assertSnapshots } from '../../../utils/TestUtils/snapshot';
import RadioGroup from './RadioGroup';

jest.mock('./RadioGroup.styles', () => ({}));
jest.mock('../RadioButton', () => 'RadioButton');

describe('RadioGroup snapshot testing', () => {
  const configs = [
    {
      props: {},
      desc: 'renders with default value'
    },
    {
      props: {
        options: [
          {
            name: 'Male',
            id: 'male'
          },
          {
            name: 'Female',
            id: 'female'
          }
        ],
        selectedIndex: 1,
        spaceBetween: 0
      },
      desc: 'renders with defined value'
    }
  ];
  assertSnapshots(RadioGroup, configs);
});
