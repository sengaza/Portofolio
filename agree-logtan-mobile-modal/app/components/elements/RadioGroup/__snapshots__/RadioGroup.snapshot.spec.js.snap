// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup snapshot testing renders with default value 1`] = `
<View
  style={undefined}
/>
`;

exports[`RadioGroup snapshot testing renders with defined value 1`] = `
<View
  style={undefined}
>
  <RadioButton
    circleStyle={undefined}
    containerStyle={false}
    index={0}
    labelStyle={undefined}
    name="Male"
    onSelect={[Function]}
    selected={false}
    value="male"
  />
  <RadioButton
    circleStyle={undefined}
    containerStyle={
      Object {
        "marginLeft": 0,
      }
    }
    index={1}
    labelStyle={undefined}
    name="Female"
    onSelect={[Function]}
    selected={true}
    value="female"
  />
</View>
`;
